beacon/blockchain/types.go

```diff
@@ -240,4 +248,18 @@ type TelemetrySink interface {
 	MeasureSince(key string, start time.Time, args ...string)
 }
 
+//nolint:revive // We will remove this once generics are cleaned up
+type BlockchainI[GenesisT any] interface {
```

Comment:
@fridrik01  it seems to me we can drop this parameter. I can compile a non-generic `BlockchainI`. Why is it here?